src/github/client/github-installation-client.ts

```diff
@@ -213,28 +213,24 @@ export class GitHubInstallationClient extends GitHubClient {
 		return response?.data?.data?.viewer?.repositories?.totalCount;
 	}
 
-	public async getBranchesPage(owner: string, repoName: string, perPage = 1, cursor?: string): Promise<getBranchesResponse> {
-		const response = await this.graphql<getBranchesResponse>(getBranchesQueryWithChangedFiles,
-			{
-				owner,
-				repo: repoName,
-				per_page: perPage,
-				cursor
-			}).catch((err) => {
-			// Is it a changedFiles error?
-			if (!isChangedFilesError(err)) {
-				return Promise.reject(err);
-			}
-
-			this.logger.warn("retrying branch graphql query without changedFiles");
-			return this.graphql<getBranchesResponse>(getBranchesQueryWithoutChangedFiles,
-				{
-					owner,
-					repo: repoName,
-					per_page: perPage,
-					cursor
-				});
-		});
+	public async getBranchesPage(owner: string, repoName: string, perPage = 1, timeCutoff?: number, cursor?: string): Promise<getBranchesResponse> {
+		const variables = {
+			owner,
+			repo: repoName,
+			per_page: perPage,
+			commitSince: timeCutoff ? new Date(Date.now() - timeCutoff).toISOString() : undefined,
```

Comment:
Shouldn't this be feature-flagged?

Reply:
Or maybe not to feature-flag but no to include the key to the object if `timeCutoffMsecs` is not provided.